### PR TITLE
fix image attachment blur deselect

### PR DESCRIPTION
**Pull request by mlsteele**
@cjb

### CODE REVIEW

File: shared/chat/conversation/container.tsx [hunk 1]
```diff
@@ -27,7 +27,9 @@ export class Conversation extends React.PureComponent<SwitchProps> {
     this.props.selectConversation()
   }
   _onWillBlur = () => {
-    this.props.deselectConversation()
+    if (Container.isPhone) {
+      this.props.deselectConversation()
```
Comment (chrisnojima) [inline shared/chat/conversation/container.tsx]: i think you want this below, but really just make this be a no op in the dispatch , we never want it to run on tablet right?

**Comment by mlsteele** [comment]
Changed to be more like desktop in 2 ways
1. Conversation disappearing does not clear `selectedConversation`
2. Inbox coming into view dispatches `tabSelected` which causes markAsRead. For the case where you've got a badge on the `selectedConversation` but are not on the chat tab and then switch to the chat tab. The badge gets cleared.

I'm still worried about [this](https://github.com/keybase/client/blob/master/shared/actions/chat2/index.tsx#L635) but not sure how to make it happen.

File: shared/constants/platform.d.ts [hunk 2]
```diff
@@ -20,6 +20,7 @@ export const isAndroidNewerThanN: boolean
 export const defaultUseNativeFrame: boolean
 export const isTestDevice: boolean
 export const isRemoteDebuggerAttached: boolean
+export const chatSplit: boolean // Whether the inbox and conversation panels are visible together.
```
Comment (chrisnojima) [inline shared/constants/platform.d.ts]: lets not put area specific things in here. this is for the platform

File: shared/util/container.tsx [hunk 3]
```diff
@@ -82,7 +82,7 @@ export const timeoutPromise = (timeMs: number) =>
   })
 
 export {default as connect, namedConnect} from './typed-connect'
-export {isMobile, isIOS, isAndroid, isPhone, isTablet} from '../constants/platform'
+export {isMobile, isIOS, isAndroid, isPhone, isTablet, chatSplit} from '../constants/platform'
```
Comment (chrisnojima) [inline shared/util/container.tsx]: shouldn't be in here

**Comment by mlsteele** [comment]
Where should it live then? I think it's conceptually helpful to switch on chatSplit instead of platform directly for routing and saga stuff. T-30m

**Comment by chrisnojima** [comment]
cc: @mmaxim

**Comment by mlsteele** [comment]
ready?

**Pull request merged**